src/neurostatslib/glm.py

```diff
@@ -1,431 +1,598 @@
-            The predicted firing rates.
+        score :
+            The log-likelihood or the pseudo-$R^2$ of the current model.
 
         Raises
         ------
         NotFittedError
             If ``fit`` has not been called first with this instance.
         ValueError
             If attempting to simulate a different number of neurons than were
-            present during fitting (i.e., if ``init_spikes.shape[0] !=
-            self.baseline_log_fr_.shape[0]``).
+            present during fitting (i.e., if ``init_y.shape[0] !=
+            self.baseline_link_fr_.shape[0]``).
 
-        See Also
-        --------
-        score
-            Score predicted firing rates against target spike counts.
-        simulate
-            Simulate spikes using GLM as a recurrent network, for extrapolating into the future.
+        Notes
+        -----
+        The log-likelihood is not on a standard scale, its value is influenced by many factors,
+        among which the number of model parameters. The log-likelihood can assume both positive
+        and negative values.
+
+        The Pseudo-$R^2$ is not equivalent to the $R^2$ value in linear regression. While both provide a measure
+        of model fit, and assume values in the [0,1] range, the methods and interpretations can differ.
+        The Pseudo-$R^2$ is particularly useful for generalized linear models where a traditional $R^2$ doesn't apply.
+
+        Refer to the `nsl.noise_model.NoiseModel` concrete subclasses for the specific likelihood equations.
+
+
+        References
+        ----------
+        [1] Cohen, Jacob, et al. Applied multiple regression/correlation analysis for the behavioral sciences.
+        Routledge, 2013.
 
         """
-        try:
-            Ws = self.spike_basis_coeff_
-        except AttributeError:
-            raise NotFittedError(
-                "This GLM instance is not fitted yet. Call 'fit' with appropriate arguments."
+        if score_type not in ["log-likelihood", "pseudo-r2"]:
+            raise NotImplementedError(
+                f"Scoring method {score_type} not implemented! "
+                f"`score_type` must be either 'log-likelihood', or 'pseudo-r2'."
             )
-        bs = self.baseline_log_fr_
-        if spike_data.shape[0] != bs.shape[0]:
+        # ignore the last time point from predict, because that corresponds to
+        # the next time step, which we have no observed data for
+        self._check_is_fit()
+        Ws = self.basis_coeff_
+        bs = self.baseline_link_fr_
+
+        X, y = self._convert_to_jnp_ndarray(X, y)
+
+        self._check_input_dimensionality(X, y)
+        self._check_input_n_timepoints(X, y)
+        self._check_input_and_params_consistency((Ws, bs), X=X, y=y)
+        if score_type == "log-likelihood":
+            norm_constant = jax.scipy.special.gammaln(y + 1).mean()
+            score = -self._score((Ws, bs), X, y) - norm_constant
+        else:
```

Comment:
Ok, I'll remove the raise that is at the beginning of the function call. The reason I first checked for the score type at the beginning is to avoid checking the input in case the score was not implemented